### PR TITLE
Revise smart contract docs

### PR DESCRIPTION
**Pull request by ceyonur**
Revised smart contract docs:
1- Truffle documentation scripts were misleading since personal API is disabled by default. Fixed it by using different script that does not use any `web3.eth.personal` calls.
2- Added Hardhat documents similar to https://github.com/ava-labs/avalanche-smart-contract-quickstart/blob/main/README.md. On top of this README, added a section to show how to interact with smart contract via hardhat console. Removed Metamask instructions as it can be confusing and it's not directly related to hardhat.

### CODE REVIEW

File: build/tutorials/smart-contracts/README.md [hunk 1]
```diff
@@ -3,6 +3,7 @@
 | Title | Description |
 | :--- | :--- |
 | [**Deploy a Smart Contract on Avalanche Using Remix and MetaMask**](deploy-a-smart-contract-on-avalanche-using-remix-and-metamask.md) | Deploy and test a smart contract on Avalanche using Remix and MetaMask. |
+| [**Using Hardhat with the Avalanche C-Chain**](using-truffle-with-the-avalanche-c-chain.md) | Using Hardhat with the Avalanche's C-Chain, an instance of the Ethereum Virtual Machine \(EVM\) |
```
Comment (jpop32) [inline build/tutorials/smart-contracts/README.md:6]: Wrong link?

**Review by ceyonur — COMMENTED** [review]
lgtm @danlaine

**Comment by danlaine** [comment]
I asked @jpop32 to look this over too. Once he gives a thumbs up I'll merge.

**Comment by jpop32** [comment]
Looks good to me.

One thing that we might add once this is up: I think there was a way to use hardhat to fork the mainnet to your local environment and then see how your contract would work with the state that is on mainnet, without actually deploying on mainnet. Unless I am wrong That would be super-useful.

**Review by cgcardona — APPROVED** [review]

**Pull request merged**